daiquiri/datalink/adapter.py

```diff
@@ -0,0 +1,165 @@
+from django.conf import settings
+from django.urls import reverse
+
+from daiquiri.core.constants import ACCESS_LEVEL_PUBLIC
+from daiquiri.core.adapter import DatabaseAdapter
+from daiquiri.core.utils import import_class, get_doi_url
+
+
+def DatalinkAdapter():
+    return import_class(settings.DATALINK_ADAPTER)()
+
+
+class BaseDatalinkAdapter(object):
+    """
+    Each datalink adapter needs to configure a set of resource types.
+
+    For each resource type, the following methods need to be implemented:
+
+    * get_<resource_type>_list(self): returns a list of all resources for the resource_type
+
+    * get_<resource_type>_identifier(self, resource): returns the identifier for
```

Comment:
I would add a little comment on what `resource` is suppose to be (a list and not a dict).

```suggestion
    * get_<resource_type>_identifier(self, resource): the parameter resource is a list of fields (cols) that represent the resource entry (row) from the DB. The method returns the fields that represents the identifier.```